packages/ddp-server/writefence.js

```diff
@@ -59,6 +60,15 @@ _.extend(DDPServer._WriteFence.prototype, {
     self._maybeFire();
   },
 
+  // Register a function to be called before firing the fence.
+  onBeforeFire: function (func) {
+    var self = this;
+    if (self.fired)
```

Comment:
Would it be correct to make this a `if (self.armed)` check instead? Then you don't have to worry about `self.before_fire_callbacks` being mutated while being iterated over.  (Actually I'm not sure why `onAllCommitted` doesn't do a `if (self.armed)` check either.)